File: Graph(그래프)/백준_1260_DFS와 BFS/DFS와 BFS.java
import java.io.*;
import java.util.*;

public class Main {

    static ArrayList<Integer>[] edge = new ArrayList[1001];
    static boolean[] check = new boolean[1001];

    static void DFS(int n){
        if(check[n] == true) return;

        check[n] = true;
        System.out.print(n + " ");
        for(int i : edge[n]){
            if(check[i] == false) DFS(i);
        }
    }

    static void BFS(int n){
        Queue<Integer> q = new LinkedList<>();
        q.add(n);
        check[n] = true;

        while(q.isEmpty() == false){
            int f = q.peek();
            q.poll();
            System.out.print(f + " ");
            for(int i : edge[f]){
                if(check[i] == false) {
                    q.add(i); check[i] = true;
                }
            }
        }
    }

    public static void main(String[] args) {
        for(int i = 0; i< 1001; i++) edge[i] = new ArrayList();
        Scanner s = new Scanner(System.in);
        int N = s.nextInt(); int M = s.nextInt(); int v = s.nextInt();

        while(M-- > 0){
            int a = s.nextInt();
            int b = s.nextInt();
            edge[a].add(b);
            edge[b].add(a);
        }

        for(int i =0 ; i<= N; i++) Collections.sort(edge[i]);
        DFS(v);
        for(int i =0 ; i<= N; i++) check[i] = false;
        System.out.println();
        BFS(v);

    }
}

File: Graph(그래프)/백준_1260_DFS와 BFS/DFS와 BFS.kt
import java.util.*
import kotlin.math.*

var edge = Array<ArrayList<Int>>(1001) {ArrayList<Int>()}
var check = BooleanArray(1001, {false})

fun DFS(n: Int){
    if(check[n] == true) return

    print("${n} ")
    check[n] = true
    for(i in edge[n]) DFS(i)
}

fun BFS(n: Int){
    var q = LinkedList<Int>()
    q.add(n)
    check[n] = true

    while(q.isEmpty() == false){
        var f = q.peek()
        q.poll()
        print("${f} ")
        for(i in edge[f]){
            if(check[i] == false){
                q.add(i)
                check[i] = true
            }
        }
    }
}

fun main(args: Array<String>){
    val s = Scanner(System.`in`)
    val N = s.nextInt()
    var M = s.nextInt()
    val v = s.nextInt()

    while(M-- >0){
        val a = s.nextInt()
        val b = s.nextInt()
        edge[a].add(b)
        edge[b].add(a)
    }

    for(i in 0..N) edge[i].sort()
    DFS(v)
    for(i in 0..N) check[i] = false
    println()
    BFS(v)

}